src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,88 @@
+    dispatch(loaderSlice.actions.setLoading(false));
+  };
+
+  useEffect(() => {
+    if (todo && !user) {
+      dispatch(loaderSlice.actions.setLoading(true));
+      getUser(todo.userId)
+        .then(res => setUser(res))
+        .finally(() => dispatch(loaderSlice.actions.setLoading(false)));
+    }
+  }, [dispatch, todo, user]);
+
   return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
-
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+    <>
+      {todo && (
+        <div
+          className={classNames('modal', { 'is-active': todo })}
+          data-cy="modal"
+        >
+          <div className="modal-background" />
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+          <div className="modal-card">
+            {loading ? (
+              <Loader />
+            ) : (
+              <>
+                <header className="modal-card-head">
+                  <div
+                    className="modal-card-title has-text-weight-medium"
+                    data-cy="modal-header"
+                  >
+                    Todo #{todo.id}
+                  </div>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+                  {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                  <button
+                    type="button"
+                    className="delete"
+                    data-cy="modal-close"
+                    onClick={handleCloseModal}
+                  />
+                </header>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+                <div className="modal-card-body">
+                  <p className="block" data-cy="modal-title">
+                    {todo.title}
+                  </p>
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+                  <p className="block" data-cy="modal-user">
+                    {!todo.completed && (
+                      <strong className="has-text-danger">Planned</strong>
+                    )}
+
+                    {todo.completed && (
+                      <strong className="has-text-success">Done</strong>
+                    )}
+
+                    {' by '}
+                    {user && <a href={`mailto:${user.email}`}>{user.name}</a>}
```

Comment:
The `href` attribute for the email link is incorrectly formatted. It should be `href={`mailto:${user.email}`}` without the extra quote to ensure the email link functions correctly.